arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Fold.kt

```diff
@@ -271,14 +271,16 @@ public annotation class DelicateRaiseApi
  * It drives the short-circuiting behavior of [Raise].
  */
 @DelicateRaiseApi
-public sealed class RaiseCancellationException(
-  internal val raised: Any?,
+public expect sealed class RaiseCancellationException(
+  raised: Any?,
+  raise: Raise<Any?>
+) : CancellationException {
+  internal val raised: Any?
   internal val raise: Raise<Any?>
-) : CancellationException(RaiseCancellationExceptionCaptured)
```

Comment:
This seems very suspicious. We need to make `RaiseCancellationException` a subclass of `CancellationException` so everything works as intended with coroutines. @nomisRev WDYT?

Reply:
This line is written exactly to provide this contract:
<img width="968" height="390" alt="image" src="https://github.com/user-attachments/assets/1f91858c-8eb1-4b7e-9e0f-5fc1e8f59436" />
Because it is `expect`, I can't provide implementation anymore. Super constructor call is implementation, so this code is written in `actual` declarations now.